code/modules/surgery/organs/dwarf_organ.dm

```diff
@@ -0,0 +1,85 @@
+/obj/item/organ/alcoholvessel //essentially the opposite of the xeno's plasmavessel, but with alcohol
+	name = "adamantine infused lungs"
+	icon_state = "plasma"
+	origin_tech = "biotech=5"
+	w_class = WEIGHT_CLASS_NORMAL
+	zone = "chest"
+	slot = "dwarf_organ"
+	var/stored_alcohol = 250
+	var/max_alcohol = 500
+	var/heal_rate = 0.5
+	var/alcohol_rate = 10
+	var/cooldown = 35
+	var/current_cooldown = 0
+/obj/item/organ/alcoholvessel/prepare_eat()
+	var/obj/S = ..()
+	S.reagents.add_reagent("ethanol", stored_alcohol/10)
+	return S
+
+
+/obj/item/organ/alcoholvessel/on_life()
+	//BEARD SENSE
+	if(current_cooldown <= world.time)
+		current_cooldown = world.time + cooldown
+		mineral_scan_pulse(get_turf(owner))
+	// MIASMA HANDLING
+	var/miasma_counter = 0
+	for(var/fuck in view(owner,7))
+		if(istype(fuck, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = fuck
+			if(H.stat == DEAD)
+				miasma_counter += 10
+		if(istype(fuck, /obj/effect/decal/cleanable/blood))
+			if(istype(fuck, /obj/effect/decal/cleanable/blood/gibs))
+				miasma_counter += 1
+			else
+				miasma_counter += 0.1
+	switch(miasma_counter)
+		if(11 to 25)
+			if(prob(5))
+				to_chat(owner, "<span class = 'danger'>Someone should really clean up in here!</span>")
+		if(26 to 50)
+			if(prob(5))
+				to_chat(owner, "<span class = 'danger'>The stench makes you queasy.</span>")
+				if(prob(5))
+					owner.vomit(20)
+		if(51 to 75)
+			if(prob(10))
+				to_chat(owner, "<span class = 'danger'>By Armok! You won't be able to keep ale down at all!</span>")
+				if(prob(10))
+					owner.vomit(20)
+		if(76 to 100)
+			if(prob(15))
+				to_chat(owner, "<span class = 'userdanger'>You can't live in such filth!</span>")
+				if(prob(15))
```

Comment:
Pretty sure a 2% chance to get stunned by vomiting and experiencing tox loss is a lot better than 15%